node/src/components/block_synchronizer.rs

```diff
@@ -519,18 +519,23 @@ impl BlockSynchronizer {
                 }
                 NeedNext::FinalitySignatures(block_hash, era_id, validators) => {
                     builder.set_in_flight_latch();
-                    results.extend(peers.into_iter().flat_map(|node_id| {
-                        validators.iter().flat_map(move |public_key| {
-                            let id = FinalitySignatureId {
-                                block_hash,
-                                era_id,
-                                public_key: public_key.clone(),
-                            };
+                    for (validator, peer) in validators
```

Comment:
Just to double check: we are losing certainty that we will get the signatures at this step since we are no longer asking multiple peers for each signature, and it becomes more likely that some of these fetches will miss. However, that will not be a problem because we will fill the missing slots in the following passes through `NeedNext::FinalitySignatures`. Is this correct?

Reply:
Yes, exactly.